### PR TITLE
setting stats interval to 100ms

### PR DESCRIPTION
**Pull request by osimhi213**
- sensitive to stream start
- fixing the cutoff bug
- https://app.asana.com/0/1207192568376365/1207237906837102/f
- @benyael15 don't merge yet, I want to check with you the stats to mixpanel..

### CODE REVIEW

File: src/createStreamingManager.ts [hunk 1]
```diff
@@ -37,57 +37,58 @@ function mapConnectionState(state: RTCIceConnectionState): ConnectionState {
...
-                        if (!isPlaying && isPlayingFalseNumIntervals === 2) {
-                            onVideoStateChange?.(StreamingState.Stop, videoStatsReport);
-                        }
-                    }
-                    videoStats.push(report);
+function createVideoStatsAnalyzer() {
+    let lastBytesReceived = 0;
+
+    return (stats: RTCStatsReport) => {
+        stats.forEach(report => {
```
Comment (HaMMeRSI) [inline src/createStreamingManager.ts:44]: can we change it to 
for (const report of stats) ?

File: src/createStreamingManager.ts [hunk 2]
```diff
@@ -37,57 +37,58 @@ function mapConnectionState(state: RTCIceConnectionState): ConnectionState {
...
+            if (report.type === 'inbound-rtp' && report.kind === 'video') {
+                if (report) {
```
Comment (HaMMeRSI) [inline src/createStreamingManager.ts:46]: if (report.type === 'inbound-rtp' && report.kind === 'video') {
                if (report) {

the order doesn't make sense

File: src/createStreamingManager.ts [hunk 3]
```diff
@@ -37,57 +37,58 @@ function mapConnectionState(state: RTCIceConnectionState): ConnectionState {
...
+                    const currBytesReceived = report.bytesReceived;
+                    const isReceiving = currBytesReceived - lastBytesReceived > 0;
+
+                    lastBytesReceived = currBytesReceived
+
+                    return isReceiving
```
Comment (HaMMeRSI) [inline src/createStreamingManager.ts:52]: ;

File: src/createStreamingManager.ts [hunk 4]
```diff
@@ -37,57 +37,58 @@ function mapConnectionState(state: RTCIceConnectionState): ConnectionState {
...
+    let lastBytesReceived = 0;
+
+    return (stats: RTCStatsReport) => {
+        stats.forEach(report => {
+            if (report.type === 'inbound-rtp' && report.kind === 'video') {
+                if (report) {
+                    const currBytesReceived = report.bytesReceived;
+                    const isReceiving = currBytesReceived - lastBytesReceived > 0;
+
+                    lastBytesReceived = currBytesReceived
```
Comment (HaMMeRSI) [inline src/createStreamingManager.ts:50]: ;

File: src/createStreamingManager.ts [hunk 5]
```diff
@@ -37,57 +37,58 @@ function mapConnectionState(state: RTCIceConnectionState): ConnectionState {
...
+            notReceivingNumIntervals = 0;
+
+            if (!isStreaming) {
+                onVideoStateChange?.(StreamingState.Start);
+
+                isStreaming = true;
+            }
+
+        } else {
+            notReceivingNumIntervals++;
```
Comment (HaMMeRSI) [inline src/createStreamingManager.ts:83]: you always increase it no?

**Review by HaMMeRSI — CHANGES_REQUESTED** [review]

File: src/createStreamingManager.ts (hunk 2 above)
Comment (osimhi213) [reply src/createStreamingManager.ts:46]: I kept the original order, you're right. I'll add it to the condition

File: src/createStreamingManager.ts (hunk 5 above)
Comment (osimhi213) [reply src/createStreamingManager.ts:83]: as long as not receiving no ? 
I zero it overhere:
if (isReceivingBytes(stats)) {
            notReceivingNumIntervals = 0;

            if (!isStreaming) {
                onVideoStateChange?.(StreamingState.Start);

                isStreaming = true;
            }

        }

Comment (osimhi213) [reply src/createStreamingManager.ts:81]: done

**Review by HaMMeRSI — APPROVED** [review]

**Pull request merged**